beta-src/src/state/game/game-api-slice.ts

```diff
@@ -344,63 +344,67 @@ const gameApiSlice = createSlice({
     },
     processUnitDoubleClick(state, clickData) {
       const {
-        order,
         data: {
           data: { contextVars },
         },
+        ownUnits,
       } = current(state);
+      if (!ownUnits.includes(clickData.payload.unitID)) {
+        return;
+      }
       if (contextVars?.context?.orderStatus) {
         const orderStates = getOrderStates(contextVars?.context?.orderStatus);
         if (orderStates.Ready) {
           return;
         }
       }
-      const { inProgress } = order;
-      if (inProgress) {
-        // double click move in progress
-      } else {
-        startNewOrder(state, clickData);
-      }
+      startNewOrder(state, clickData);
     },
     processUnitClick(state, clickData) {
       const {
-        order,
         data: {
           data: { contextVars },
         },
+        order: {
+          inProgress,
+          method,
+          onTerritory,
+          orderID,
+          toTerritory,
+          type,
+          unitID,
+        },
+        ownUnits,
       } = current(state);
       if (contextVars?.context?.orderStatus) {
         const orderStates = getOrderStates(contextVars?.context?.orderStatus);
         if (orderStates.Ready) {
           return;
         }
       }
-      const { inProgress } = order;
       if (inProgress) {
-        if (order.type === "hold" && order.onTerritory !== null) {
+        if (unitID === clickData.payload.unitID) {
+          resetOrder(state);
+        } else if (type === "hold" && onTerritory !== null) {
           highlightMapTerritoriesBasedOnStatuses(state);
-        } else if (order.type === "move" && order.toTerritory !== null) {
+        } else if (type === "move" && toTerritory !== null) {
           highlightMapTerritoriesBasedOnStatuses(state);
         } else if (
```

Comment:
This else if statement is doing the same thing for both checks. Can we do an `||` statement in the first check to include `type === "move" && toTerritory !== null`